tutorial/westeros/westeros_addon_technologies.ipynb

```diff
@@ -0,0 +1,607 @@
+    "\n",
+    "This tutorial is concerned with is looking at how to add additional features, known as `addon` technologies such as carbon-capture-and-storage (CCS) retrofits, passout-turbines (for optional heat cogeneration) or cooling technologies, to existing technologies.\n",
+    "\n",
+    "There are several ways in which to tackle this issue.  Lets take for example our `coal_ppl`.  All of the above mentioned additional features could be implemented by introducing different modes for the `coal_ppl`.  For example, heat cogeneration could be implemented as a separate `mode` of the `coal_ppl`, where instead of just generating electricity, heat can also be produced at the cost of reducing the amount of electricity generated. Another approach would make use of the generic `relations`, therefore linking the a newly added technology representing the passout-turbine with the activity of the `coal_ppl`.  Both of these approaches have some downsides. Using a separate `mode` will not permit explicitly modelling investment costs and lifetimes associated with the asset being added to the `coal_ppl`.  Generic relations are very flexible, but if too many of these exist, then the modell quickly becomes very hard to understand.  Therefore, MESSAGEix offers an explicit `addon` formulation.\n",
+    "\n",
+    "What this means, is that the additional technology options are explicitly modelled as separate technologies, classified as `addon` technologies, linked to the activity of the technology to which they serve as additional configuration options, the parent technology, through specifying a `addon_conversion` factor.  The activity of the `addon` technology can further be restricted to a minimum or maximum share of the activity of the parent technology."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adding heat co-generation capacibility to the `coal_ppl`\n",
+    "This tutorial will extend the current reference-energy-system to include a demand for heat and the necessary technologies to meet this demand.  Heat will be generated via a `passout-turbine` which will be linked to the `coal_ppl` using the `addon` formulation.\n",
+    "\n",
+    "# *Add modified RES*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import ixmp\n",
+    "import message_ix\n",
+    "\n",
+    "from message_ix.utils import make_df\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mp = ixmp.Platform()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load the existing scenario '*baseline*' and clone to a new scenario '*addon_technology*' which we will extended to include heat cogeneration using the `addon` feature"
+   ]
```

Comment:
Please convert long titles/headings to short and concise ones, and bring the text below the heading.